### DISCUSSION TITLE
(datepicker) Ability to enter time picker values via keyboard input

### DISCUSSION THREAD

**Discussion by ashnewport** [matheo/angular#16] [Feedback]
Love the datepicker library 👍

I was trying to enter a date/time into a field and kept getting unexpected results or `null`. Let's say I wanted to manually enter the time as `03:00 AM` via the keyboard, when I type in the value it returns invalid. I think I understand how this is happening in that the validation or setting of the fields value is expecting a full datetime string in the format close to `month/day/year, hour:minute meridian`. So entering a value starting with `03:` will not be considered a valid month value.

The feature request I'd like to discuss is to enable keyboard input to be allowed and valid with context to the type of input expected. E.g. time should be allowed to be entered in time fields, date should be allowed to be entered in date fields.

The demos at http://matheo.co/demos/datepicker showcase current behaviour.

For example:

- type `6` into the 'TIME PICKER' input and observe the value below outputs as `12:00 AM`
- type `6` into the 'DATETIME PICKER' input and observe the value below outputs as `6/1/01, 12:00 AM`

**Comment by matheo** [comment]
Hi @ashnewport 
Indeed, I did not implementing anything in that matter, I'm not sure which handlers are taking care of Keyboard events in the Click view, and unfortunately I have stronger priorities with `@myndpm/dyn-forms`.

Unless we got a PR solving this issue, it's going to wait some time :(
